Use the FinalOutputPath instead of the Identity of output items

joelverhagen opened the pull request:
The targets `BuiltProjectOutputGroup` and `DocumentationProjectOutputGroup` produce the output items.

Fixes NuGet/Home#3880.

The `Identity` of the item points to `obj` for `BuiltProjectOutputGroup`.

/cc @emgarten @mishra14 @rainersigwald @nguerrera

rainersigwald reviewed (approved):
Another possible approach would be to use a transform on the [input](https://github.com/NuGet/NuGet.Client/blob/5c063150958167022df676a10829240cb2e12e95/src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets#L88) to this task, which would slightly decouple the task from the return format of `BuiltProjectOutputGroup`. But since the task and the target will rev together, I don't think there's a strong need to move to that approach, so :shipit:.

joelverhagen commented:
🔔 can haz reviews? 🐱

rainersigwald reviewed (approved):
Nice. It's a bit cleaner this way, I'd say.

emgarten commented on src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs:
```diff
@@ -115,6 +115,19 @@ private PackArgs GetPackArgs()
+            }
+
+            return TargetPathsToAssemblies
+                .Where(path => !string.IsNullOrEmpty(path))
```
string.IsNullOrWhitespace might be safer since msbuild doesn't trim extra new lines characters

emgarten replied:
and each value should be trimmed

joelverhagen replied:
Fixed.

joelverhagen committed:
Use the FinalOutputPath instead of the Identity of output items

The targets BuiltProjectOutputGroup and DocumentationProjectOutputGroup produce the output items
Fix https://github.com/NuGet/Home/issues/3880
